packages/rocketchat-api/server/v1/groups.js

```diff
@@ -308,6 +308,32 @@ RocketChat.API.v1.addRoute('groups.list', { authRequired: true }, {
 	}
 });
 
+RocketChat.API.v1.addRoute('groups.listAll', { authRequired: true }, {
+	get() {
+		const { offset, count } = this.getPaginationItems();
+		const { sort, fields } = this.parseJsonQuery();
+		if (!RocketChat.authz.hasPermission(this.userId, 'view-room-administration')) {
```

Comment:
Thanks for making the changes, any chance you can move this permission check to be above the other two items?